ansible/roles/dev/tasks/main.yml

```diff
@@ -17,20 +17,39 @@
   notify: restart sshd
 
 - name: Install python-virtualenvwrapper
-  package: name=python-virtualenvwrapper state=latest
+  package:
+    name: "{{ item }}"
+    state: latest
+  with_items:
+    # FIXME(milan): a dependency bug?
+    - which
```

Comment:
I'm surprised this is necessary. If `which` is a dep of virtualenvwrapper, it should be installed then.

Reply:
I guess the `python[23]-virtualenvwrapper` packages miss the `which` dependency:
Is `which` required in any alias?

```
(pulp) [vagrant@pulp3 pulp]$ set | grep -C3 which
+ grep --color=auto -C3 which
+ set
}
virtualenvwrapper_verify_resource ()
{
    typeset exe_path="$(command \which "$1" | (unset GREP_OPTIONS; command \grep -v "not found"))";
    if [ "$exe_path" = "" ]; then
        echo "ERROR: virtualenvwrapper could not find $1 in your path" 1>&2;
        return 1;
++ printf '\033]0;%s@%s:%s\007' vagrant pulp3 '~/devel/pulp'
```

 Create a custom function to test it:
```
(pulp) [vagrant@pulp3 pulp]$ function f () {
> typeset exe_path="$(command \which "$1" | (unset GREP_OPTIONS; command \grep -v "not found"))";
> }

(pulp) [vagrant@pulp3 pulp]$ f python
+ f python
++ command which python
++ unset GREP_OPTIONS
++ command grep -v 'not found'
++ grep -v 'not found'
++ which python
+ typeset exe_path=/home/vagrant/.virtualenvs/pulp/bin/python
++ printf '\033]0;%s@%s:%s\007' vagrant pulp3 '~/devel/pulp'
(pulp) [vagrant@pulp3 pulp]$ f foo
+ f foo
++ command which foo
++ which foo
++ unset GREP_OPTIONS
++ command grep -v 'not found'
++ grep -v 'not found'
which: no foo in (/home/vagrant/.virtualenvs/pulp/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/vagrant/.local/bin:/home/vagrant/bin)
+ typeset exe_path=
++ printf '\033]0;%s@%s:%s\007' vagrant pulp3 '~/devel/pulp'
(pulp) [vagrant@pulp3 pulp]$

```

List of deps of virtualenvwrapper:

```
pulp) [vagrant@pulp3 pulp]$ rpm -q -R python2-virtualenvwrapper
/bin/sh
config(python2-virtualenvwrapper) = 4.7.2-3.fc26
python(abi) = 2.7
python2-stevedore
python2-virtualenv
python2-virtualenv-clone
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PartialHardlinkSets) <= 4.0.4-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1
(pulp) [vagrant@pulp3 pulp]$
```
`which` is actually provided by the `which` package:
```
(pulp) [vagrant@pulp3 pulp]$ dnf whatprovides '*bin/which'
Last metadata expiration check: 0:12:43 ago on Thu 11 Jan 2018 04:56:01 PM UTC.
which-2.21-2.fc26.x86_64 : Displays where a particular program in your path is located
Repo        : @System
Matched from:
Other       : *bin/which
```